lib/app_config.ex

```diff
@@ -132,21 +133,41 @@ defmodule AppConfig do
   ## Example
 
       {:ok, "VALUE"} = #{inspect __MODULE__}.fetch_env(:my_app, :test_var)
+      :ok = Application.put_env(:my_test_app, :test_var_1, %{ test_var_2: "VALUE" })
+      {:ok, "VALUE"} = #{inspect __MODULE__}.fetch_env(:my_app, [ :test_var_1, :test_var_2 ])
 
   """
-  @spec fetch_env(app | Keyword.t, key) :: {:ok, value} | :error
-  def fetch_env(env, key) when (is_atom(env) or is_list(env)) and is_atom(key) do
+  @spec fetch_env(app | Keyword.t, key | List.t) :: {:ok, value} | :error
+  def fetch_env(env, key) when (is_atom(env) or is_list(env) or is_map(env)) and is_atom(key) do
     with {:ok, value} <- fetch(env, key) do
       get_env_value(value)
     end
   end
+  def fetch_env(env, [ first_key | keys ]) when is_atom(env) or is_list(env) or is_map(env) do
+    with {:ok, value} <- Enum.reduce(keys, fetch(env, first_key), &fetch_nested/2) do
+      get_env_value(value)
+    end
+  end
+
+  defp fetch_nested(key, { :ok, val }) do
+    case val do
+      val when is_map(val) or is_list(val) -> fetch(val, key)
+      _ -> :error
+    end
+  end
+  defp fetch_nested(_, :error) do
+    :error
+  end
```

Comment:
The `fetch_nested/2` function could be simplified a bit. Maybe to something like:
```elixir
defp fetch_nested(key, {:ok, val}) when is_map(val) or is_list(val) do
  fetch(val, key)
end
defp fetch_nested(_key, _result) do
  :error
end
```